decode_test.go

```diff
@@ -868,16 +868,17 @@ var unmarshalErrorTests = []struct {
 	{"b: *a\na: &a {c: 1}", `yaml: unknown anchor 'a' referenced`},
 	{"%TAG !%79! tag:yaml.org,2002:\n---\nv: !%79!int '1'", "yaml: did not find expected whitespace"},
 	{"a:\n  1:\nb\n  2:", ".*could not find expected ':'"},
+	{"a:\n  1: b\nc\n\n\n\n\n\n", ".*line 2: could not find expected ':'"},
```

Comment:
This test is used to validate the change.

Without the change, one gets:
`"yaml: line 9: could not find expected ':'"`